File: src/main/java/main/commands/slashcommands/commands/management/permissions/ListDJCommand.java
package main.commands.slashcommands.commands.management.permissions;

import main.commands.prefixcommands.CommandContext;
import main.commands.prefixcommands.ICommand;
import main.constants.Permission;
import main.utils.GeneralUtils;
import main.utils.RobertifyEmbedUtils;
import main.utils.component.interactions.AbstractSlashCommand;
import main.utils.locale.LocaleManager;
import main.utils.locale.RobertifyLocaleMessage;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.internal.utils.tuple.Pair;
import org.jetbrains.annotations.NotNull;

import javax.script.ScriptException;
import java.util.List;

public class ListDJCommand extends AbstractSlashCommand implements ICommand {
    @Override
    public void handle(CommandContext ctx) throws ScriptException {
        if (!GeneralUtils.hasPerms(ctx.getGuild(), ctx.getMember(), Permission.ROBERTIFY_DJ)) return;
        new PermissionsCommand().list(ctx.getMessage(), List.of("list", "robertify_dj"));
    }

    @Override
    public String getName() {
        return "listdj";
    }

    @Override
    public String getHelp(String prefix) {
        return "Aliases: `"+ GeneralUtils.listToString(getAliases()) +"`\n\n" +
                "Looking to see who all the DJs are for this server? This is the command to use!";
    }

    @Override
    public List<String> getAliases() {
        return List.of("ldj", "djs");
    }

    @Override
    protected void buildCommand() {
        setCommand(
                getBuilder()
                        .setName("listdj")
                        .setDescription("View all the DJs in this server")
                        .setDJOnly()
                        .build()
        );
    }

    @Override
    public String getHelp() {
        return null;
    }

    @Override
    public void onSlashCommand(@NotNull SlashCommandEvent event) {
        if (!checks(event)) return;

        final var guild = event.getGuild();
        final var localeManager = LocaleManager.getLocaleManager(guild);

        List<String> roles = new PermissionsCommand().getRolePerms(event.getGuild(), Permission.ROBERTIFY_DJ.name());
        List<String> users = new PermissionsCommand().getUserPerms(event.getGuild(), Permission.ROBERTIFY_DJ.name());

        EmbedBuilder eb = RobertifyEmbedUtils.embedMessage(guild, RobertifyLocaleMessage.PermissionsMessages.PERMISSION_LIST, Pair.of("{permission}", Permission.ROBERTIFY_DJ.name()))
                .addField(localeManager.getMessage(RobertifyLocaleMessage.PermissionsMessages.PERMISSIONS_ROLES), roles.isEmpty() ? localeManager.getMessage(RobertifyLocaleMessage.GeneralMessages.NOTHING_HERE) : GeneralUtils.listToString(roles), false)
                .addField(localeManager.getMessage(RobertifyLocaleMessage.PermissionsMessages.PERMISSIONS_USERS), users.isEmpty() ? localeManager.getMessage(RobertifyLocaleMessage.GeneralMessages.NOTHING_HERE) : GeneralUtils.listToString(users), false);
        event.replyEmbeds(eb.build())
                .setEphemeral(false)
                .queue();
    }
}

File: src/main/kotlin/main/commands/slashcommands/management/permissions/ListDJCommand.kt
package main.commands.slashcommands.management.permissions

import main.constants.RobertifyPermission
import main.utils.RobertifyEmbedUtils.Companion.replyEmbed
import main.utils.component.interactions.slashcommand.AbstractSlashCommand
import main.utils.component.interactions.slashcommand.models.SlashCommand
import main.utils.json.permissions.PermissionsConfig
import main.utils.locale.LocaleManager
import main.utils.locale.messages.GeneralMessages
import main.utils.locale.messages.PermissionsMessages
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

class ListDJCommand : AbstractSlashCommand(SlashCommand(
    name = "listdj",
    description = "List all DJ users and roles in this server.",
    adminOnly = true
)) {

    override fun handle(event: SlashCommandInteractionEvent) {
        val guild = event.guild!!
        val config = PermissionsConfig(guild)
        val localeManager = LocaleManager[guild]

        val roles = PermissionsCommand().getRolesForPermission(RobertifyPermission.ROBERTIFY_DJ, config)
        val users = PermissionsCommand().getUsersForPermission(RobertifyPermission.ROBERTIFY_DJ, config)

        event.replyEmbed {
            embedBuilder(
                PermissionsMessages.PERMISSION_LIST,
                Pair("{permission}", RobertifyPermission.ROBERTIFY_DJ.name)
            )
                .addField(
                    localeManager.getMessage(PermissionsMessages.PERMISSIONS_ROLES),
                    if (roles.isEmpty()) localeManager.getMessage(GeneralMessages.NOTHING_HERE) else roles.joinToString(", "),
                    false
                )
                .addField(
                    localeManager.getMessage(PermissionsMessages.PERMISSIONS_USERS),
                    if (users.isEmpty()) localeManager.getMessage(GeneralMessages.NOTHING_HERE) else users.joinToString(", "),
                    false
                )
                .build()
        }.queue()
    }
}